src/lib/data/alignment.js

```diff
@@ -78,49 +116,99 @@ export default class Alignment {
     return true
   }
 
+  /**
+   * @returns { AlignedText | Null } - origin aligned text
+   */
   get originAlignedText () {
-    return this.origin.alignedText
+    return this.origin.alignedText ? this.origin.alignedText : null
   }
 
+  /**
+   * @returns { AlignedText | Null } - target aligned text
+   */
   get targetAlignedText () {
-    return this.target.alignedText
+    return this.target.alignedText ? this.target.alignedText : null
   }
 
+  /**
+   * Defines if the given token should finish the active alignment group - token is already in the group and
+   * its textType is the same as token that created/activated the group
+   * @param {Token} token
+   * @returns {Boolean}
+   */
   shouldFinishAlignmentGroup (token) {
-    return this.tokenInActiveGroup(token) && this.tokenTheSameTextTypeAsStart(token)
+    return this.tokenInActiveGroup(token) && this.tokenTheSameTextTypeAsStart(token) && this.activeAlignmentGroup.couldBeFinished
   }
 
+  /**
+   * Defines if the given token should be removed from the active alignment group - token is already in the group and
+   * its textType is NOT the same as token that created/activated the group
+   * @param {Token} token
+   * @returns {Boolean}
+   */
   shouldBeRemovedFromAlignmentGroup (token) {
     return this.tokenInActiveGroup(token) && !this.tokenTheSameTextTypeAsStart(token)
   }
 
+  /**
+   * Defines if a new alignment group should be created
+   * @returns {Boolean}
+   */
   shouldStartNewAlignmentGroup () {
-    return !this.activeAlignmentGroup
+    return !this.hasActiveAlignment
+  }
+
+  /**
+   * Defines if there is an active alignment group
+   * @returns {Boolean}
+   */
+  get hasActiveAlignment () {

```

Comment:
I still see direct references to this.activeAlignmentGroup elsewhere. Did you mean to update them to use this method instead?

Reply:
Yes, I missed it - thank you for the catch! Fixed